apitools/base/py/encoding.py

```diff
@@ -307,7 +307,8 @@ def decode_field(self, field, value):
             value = result.value
             if result.complete:
                 return value
-        if isinstance(field, messages.MessageField):
+        if isinstance(field, messages.MessageField) and not isinstance(
```

Comment:
Thanks for adding this fix, great find! Are there any other classes that inherit from Message other than DateTimeField we should watch for here?